itest/rust/src/rect2_test.rs

```diff
@@ -0,0 +1,102 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+use std::fmt::Debug;
+
+use crate::itest;
+use godot::prelude::{inner::InnerRect2, *};
+
+#[itest]
+fn rect2_equiv_unary() {
+    let test_rects = [
+        Rect2::from_components(0.2, 0.3, 1.5, 0.9),
+        Rect2::from_components(0.2, 0.3, 1.5, 1.9),
+        Rect2::from_components(0.2, 0.3, 1.0, 1.9),
+        Rect2::from_components(4.2, 4.3, 1.5, 1.9),
+        Rect2::from_components(8.2, 8.3, 2.5, 2.9),
+        Rect2::from_components(8.2, 8.3, 2.5, 3.9),
+    ];
+    let test_vectors = [
+        Vector2::ZERO,
+        Vector2::new(0.0, 10.0),
+        Vector2::new(10.0, 0.0),
+        Vector2::new(10.0, 10.0),
+    ];
+    let test_reals = [0.0, 1.0, 10.0, 32.0];
+    let test_sides = [
+        RectSide::Left,
+        RectSide::Top,
+        RectSide::Right,
+        RectSide::Bottom,
+    ];
+
+    fn evaluate_mappings<T>(key: &str, a: T, b: T)
+    where
+        T: PartialEq + Debug,
+    {
+        assert_eq!(a, b, "{}: outer != inner ({:?} != {:?})", key, a, b);
```

Comment:
```suggestion
        assert_eq!(a, b, "{key}: outer != inner ({a:?} != {b:?})");
```
The clippy lints on this one are flaky, still haven't found out why.